test/syscalls/linux/epoll.cc

```diff
@@ -496,6 +496,50 @@ TEST(EpollTest, PipeReaderHupAfterWriterClosed) {
   EXPECT_EQ(result[0].data.u64, kMagicConstant);
 }
 
+TEST(EpollTest, DoubleLayerEpoll) {
+  int listen_fd = socket(AF_INET, SOCK_STREAM, IPPROTO_TCP);
```

Comment:
Create socket like this and capture in a `FileDescriptor` which would clean up after the scope ends.

https://cs.opensource.google/gvisor/gvisor/+/master:test/syscalls/linux/packet_socket_raw.cc;l=168-170;drc=dfb3273f887a80c25a6d133fd5a082153ba58570